clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -6936,7 +6945,8 @@ static void ProcessDeclAttribute(Sema &S, Scope *scope, Decl *D,
   // which do not apply to the current target architecture are treated as
   // though they were unknown attributes.
   if (AL.getKind() == ParsedAttr::UnknownAttribute ||
-      !AL.existsInTarget(S.Context.getTargetInfo())) {
+      !(AL.existsInTarget(S.Context.getTargetInfo()) ||
+        isAttrValidForSYCLDevice(S.Context, AL))) {
```

Comment:
Can't the check for DLLIMPORT/DLLEXPORT be added as part of AL.existsInTarget()?
I mean instead of adding a new func isAttrValidForSYCLDevice(), may those checks be added to existsInTarget?

Reply:
I think about that at beginning,  Can we turn off warning for all this spir/spir64 target?  At time, Alex think, we should only turn off warning for sycl-divice.   

I know openmp using those target(spir/spir64) too.  I am not sure we can turn off warning for that.

Reply:
I mean, fix the warning ONLY for sycl-device, but add the check to existsInTarget() if possible.

Reply:
the existsInTarget() is all about target, it defined as:
bool ParsedAttr::existsInTarget(const TargetInfo &Target) const {
       return getInfo(*this).ExistsInTarget(Target);
}
it is called for all attributes.  For dllimport/export:
getInfo(*this).ExistsInTraget ==>isTargetx86x86)64armthumbaarch6win32.

And that function is compiler generated by table-gen.
If we want to turn of by specified target, we can do that in existsInTraget.   I don't think we can do this in just sycl-device.

Reply:
Just wonder if sycl-device can be treat ad a specified target?

Reply:
Ok, I skip my question. I did not notice that that existsInTarget() accepts (S.Context.Target), while isAttrValidForSyclDevice() accepts S.Context.
Answering the question..., no sycl-device cannot be treated as a target.